### PR TITLE
Events

### PR DESCRIPTION
**Pull request by mrkamin**
What does this PR do?
This PR is raised to submit for code review in the Microvers dashboard of Mohammad Rafi Amin.

Description of the task to be completed 👎

I will be building my "Math magicians" website for more than a whole week. below are the tasks that will be done.
Project 1: Setup.
Project 2: Components.
**Project 3: Events.**
Project 4: Refactor with hooks.
Project 5: Full website.
Project 6: Tests (pair programming).
Exercise: Deploy.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+const root = ReactDOM.createRoot(document.getElementById('root'));
+root.render(
+  <React.StrictMode>
+    <App />
+  </React.StrictMode>,
+);
+
+// If you want to start measuring performance in your app, pass a function
+// to log results (for example: reportWebVitals(console.log))
+// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
```
Comment (OLIPLICHE) [inline src/index.js]: - I would advise you to remove these commented lines because they make your code not clean.

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import './App.css';
+import Calculator from './Component/Calculator';
+
+// eslint-disable-next-line react/prefer-stateless-function
```
Comment (OLIPLICHE) [inline src/App.js]: - I would advise you to solve this linter Manually. Linters are tools that you use to debug your code. They scan your scripts for common issues and errors.

File: src/Logic/calculate.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+/**
+ * Given a button name and a calculator data object, return an updated
+ * calculator data object.
+ *
+ * Calculator data object contains:
+ *   total:s      the running total
+ *   next:String       the next number to be operated on with the total
+ *   operation:String  +, -, etc.
+ */
```
Comment (OLIPLICHE) [inline src/Logic/calculate.js]: - Clean the commented code on this file too 👍

File: src/Logic/operate.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    }
+  }
+  if (operation === '%') {
+    try {
+      return one.mod(two).toString();
+    } catch (err) {
+      return "Can't find modulo as can't divide by 0.";
+    }
+  }
+  throw Error(`Unknown operation '${operation}'`);
```
Comment (OLIPLICHE) [inline src/Logic/operate.js:29]: Well done 👍 
Capture

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @mrkamin , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- App is working fine 👍 
- Linter is green (the last commit have a green check mark (✅)) 👏
- README file is descriptive 👏
- Commit messages are meaningful 👏 👏
- Gitflow is correctly used (Pull Request is made from the feature branch into the dev branch) 👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (mrkamin) [reply src/index.js]: @OLIPLICHE 
thanks for your feedback I will consider the points in my future task for now it solved

File: src/App.js (hunk 2 above)
Comment (mrkamin) [reply src/App.js]: @OLIPLICHE 
I can solve this by chnageing the class component to functional component but in my previous taske the code reviewer wants me to change the functional component to class component I hope my changes will approved.

File: src/Logic/calculate.js (hunk 3 above)
Comment (mrkamin) [reply src/Logic/calculate.js]: @OLIPLICHE
thanks for your feedback I will consider the points in my future task for now it solved

File: src/Logic/operate.js (hunk 4 above)
Comment (mrkamin) [reply src/Logic/operate.js:29]: @OLIPLICHE
thanks for your feedback

**Review by mahbubul14 — APPROVED** [review]
# Project Approved :trophy: :tada: 

Hi @mrkamin, 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linter error.
:heavy_check_mark: Descriptive readme file.
:heavy_check_mark: Well-implemented suggestions.

### Optional suggestions :arrow_down_small: 

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

* N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by mrkamin** [comment]
> # Project Approved 🏆 🎉
> Hi @mrkamin,
> 
> Your project is complete! You have done it exceptionally well. it's time to merge it 💯 Congratulations! 🎉
> 
> ### To Highlight 👍
> ✔️ No linter error. ✔️ Descriptive readme file. ✔️ Well-implemented suggestions.
> 
> ### Optional suggestions 🔽
> Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.
> 
> * N/A
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

thanks a lot

**Pull request merged**